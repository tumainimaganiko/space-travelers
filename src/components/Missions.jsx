import { useSelector } from 'react-redux';
import { v4 as uuidv4 } from 'uuid';
import MissionDetails from './MissionDetails';
import Styles from '../styles/Mission.module.css';

const Missions = () => {
  const { missions } = useSelector((store) => store.missions);

  return (
    <table className={Styles.missionTable}>
      <thead>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
        </tr>
      </thead>
      <tbody>
        {missions.length > 0
          ? missions.map((mission) => (
            <MissionDetails
              key={uuidv4()}
              name={mission.mission_name}
              description={mission.description}
              id={mission.mission_id}
              reserved={mission.reserved}
            />
          ))
          : 'loading'}
      </tbody>
    </table>
  );
};

export default Missions;
