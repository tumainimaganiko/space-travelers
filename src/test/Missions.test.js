import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import axios from 'axios';
import renderer from 'react-test-renderer';
import Missions from '../components/Missions';
import myMissions from './missionsData';

jest.mock('axios');

const reducer = (state = { missions: { missions: myMissions } }) => state;

const store = configureStore({ reducer });

describe('Should render correctly', () => {
  it('It should get all mission loaded', async () => {
    axios.get.mockResolvedValue({ data: myMissions });
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
