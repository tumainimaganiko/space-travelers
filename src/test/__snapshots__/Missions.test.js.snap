// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly It should get all mission loaded 1`] = `
<table
  className="missionTable"
>
  <thead>
    <tr>
      <th>
        Mission
      </th>
      <th>
        Description
      </th>
      <th>
        Status
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      style={
        Object {
          "backgroundColor": "#E5E4E2",
        }
      }
    >
      <td
        className="name"
      >
        Thaicom
      </td>
      <td
        className="description"
      >
        Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
      </td>
      <td>
        <span
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#36454F",
              "border": "1px solid gray",
              "color": "white",
              "display": "block",
              "width": "120px",
            }
          }
        >
           
          Not a Member
        </span>
      </td>
      <td>
        <button
          className="joinBtn"
          onClick={[Function]}
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
    <tr
      style={
        Object {
          "backgroundColor": "#E5E4E2",
        }
      }
    >
      <td
        className="name"
      >
        Telstar
      </td>
      <td
        className="description"
      >
        Telstar 19V (Telstar 19 Vantage) is a communication satellite in the Telstar series of the Canadian satellite communications company Telesat. It was built by Space Systems Loral (MAXAR) and is based on the SSL-1300 bus. As of 26 July 2018, Telstar 19V is the heaviest commercial communications satellite ever launched, weighing at 7,076 kg (15,600 lbs) and surpassing the previous record, set by TerreStar-1 (6,910 kg/15230lbs), launched by Ariane 5ECA on 1 July 2009.
      </td>
      <td>
        <span
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#36454F",
              "border": "1px solid gray",
              "color": "white",
              "display": "block",
              "width": "120px",
            }
          }
        >
           
          Not a Member
        </span>
      </td>
      <td>
        <button
          className="joinBtn"
          onClick={[Function]}
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
    <tr
      style={
        Object {
          "backgroundColor": "#E5E4E2",
        }
      }
    >
      <td
        className="name"
      >
        Iridium NEXT
      </td>
      <td
        className="description"
      >
        In 2017, Iridium began launching Iridium NEXT, a second-generation worldwide network of telecommunications satellites, consisting of 66 active satellites, with another nine in-orbit spares and six on-ground spares. These satellites will incorporate features such as data transmission that were not emphasized in the original design. The constellation will provide L-band data speeds of up to 128 kbit/s to mobile terminals, up to 1.5 Mbit/s to Iridium Pilot marine terminals, and high-speed Ka-band service of up to 8 Mbit/s to fixed/transportable terminals. The next-generation terminals and service are expected to be commercially available by the end of 2018. However, Iridium's proposed use of its next-generation satellites has raised concerns the service will harmfully interfere with GPS devices. The satellites will incorporate a secondary payload for Aireon, a space-qualified ADS-B data receiver. This is for use by air traffic control and, via FlightAware, for use by airlines. A tertiary payload on 58 satellites is a marine AIS ship-tracker receiver, for Canadian company exactEarth Ltd. Iridium can also be used to provide a data link to other satellites in space, enabling command and control of other space assets regardless of the position of ground stations and gateways.
      </td>
      <td>
        <span
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#36454F",
              "border": "1px solid gray",
              "color": "white",
              "display": "block",
              "width": "120px",
            }
          }
        >
           
          Not a Member
        </span>
      </td>
      <td>
        <button
          className="joinBtn"
          onClick={[Function]}
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
    <tr
      style={
        Object {
          "backgroundColor": "#E5E4E2",
        }
      }
    >
      <td
        className="name"
      >
        Commercial Resupply Services
      </td>
      <td
        className="description"
      >
        Commercial Resupply Services (CRS) are a series of contracts awarded by NASA from 2008–2016 for delivery of cargo and supplies to the International Space Station (ISS) on commercially operated spacecraft. The first CRS contracts were signed in 2008 and awarded $1.6 billion to SpaceX for 12 cargo transport missions and $1.9 billion to Orbital Sciences for 8 missions, covering deliveries to 2016. In 2015, NASA extended the Phase 1 contracts by ordering an additional three resupply flights from SpaceX and one from Orbital Sciences. After additional extensions late in 2015, SpaceX is currently scheduled to have a total of 20 missions and Orbital 10. SpaceX began flying resupply missions in 2012, using Dragon cargo spacecraft launched on Falcon 9 rockets from Space Launch Complex 40 at Cape Canaveral Air Force Station, Cape Canaveral, Florida. Orbital Sciences began deliveries in 2013 using Cygnus spacecraft launched on the Antares rocket from Launch Pad 0A at the Mid-Atlantic Regional Spaceport (MARS), Wallops Island, Virginia. A second phase of contracts (known as CRS2) were solicited and proposed in 2014. They were awarded in January 2016 to Orbital ATK, Sierra Nevada Corporation, and SpaceX, for cargo transport flights beginning in 2019 and expected to last through 2024.
      </td>
      <td>
        <span
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#36454F",
              "border": "1px solid gray",
              "color": "white",
              "display": "block",
              "width": "120px",
            }
          }
        >
           
          Not a Member
        </span>
      </td>
      <td>
        <button
          className="joinBtn"
          onClick={[Function]}
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
    <tr
      style={
        Object {
          "backgroundColor": "#E5E4E2",
        }
      }
    >
      <td
        className="name"
      >
        SES
      </td>
      <td
        className="description"
      >
        SES S.A. is a communications satellite owner and operator providing video and data connectivity worldwide to broadcasters, content and internet service providers, mobile and fixed network operators, governments and institutions, with a mission to “connect, enable, and enrich”. SES operates more than 50 geostationary orbit satellites and 16 medium Earth orbit satellites. These comprise the well-known European Astra TV satellites, the O3b data satellites and others with names including AMC, Ciel, NSS, Quetzsat, YahSat and SES.
      </td>
      <td>
        <span
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#36454F",
              "border": "1px solid gray",
              "color": "white",
              "display": "block",
              "width": "120px",
            }
          }
        >
           
          Not a Member
        </span>
      </td>
      <td>
        <button
          className="joinBtn"
          onClick={[Function]}
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
    <tr
      style={
        Object {
          "backgroundColor": "#E5E4E2",
        }
      }
    >
      <td
        className="name"
      >
        JCSAT
      </td>
      <td
        className="description"
      >
        The JSAT constellation is a communication and broadcasting satellite constellation formerly operated by JSAT Corporation and currently by SKY Perfect JSAT Group. It has become the most important commercial constellation in Japan, and the fifth of the world. It has practically amalgamated all private satellite operators in Japan, with only B-SAT left as a local competitor.
      </td>
      <td>
        <span
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#36454F",
              "border": "1px solid gray",
              "color": "white",
              "display": "block",
              "width": "120px",
            }
          }
        >
           
          Not a Member
        </span>
      </td>
      <td>
        <button
          className="joinBtn"
          onClick={[Function]}
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
    <tr
      style={
        Object {
          "backgroundColor": "#E5E4E2",
        }
      }
    >
      <td
        className="name"
      >
        AsiaSat
      </td>
      <td
        className="description"
      >
        Asia Satellite Telecommunications Holdings Limited known as its brand name AsiaSat is a commercial operator of communication spacecraft. AsiaSat is based in Hong Kong but incorporated in Bermuda.
      </td>
      <td>
        <span
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#36454F",
              "border": "1px solid gray",
              "color": "white",
              "display": "block",
              "width": "120px",
            }
          }
        >
           
          Not a Member
        </span>
      </td>
      <td>
        <button
          className="joinBtn"
          onClick={[Function]}
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
    <tr
      style={
        Object {
          "backgroundColor": "#E5E4E2",
        }
      }
    >
      <td
        className="name"
      >
        Orbcomm OG2
      </td>
      <td
        className="description"
      >
        Orbcomm Generation 2 (OG2) second-generation satellites are intended to supplement and eventually replace the current first generation constellation. Eighteen satellites were ordered by 2008—nominally intended to be launched in three groups of six during 2010–2014—and by 2015 have all been launched, on three flights. Orbcomm has options for a further thirty OG2 spacecraft. The satellites were launched by SpaceX on the Falcon 9 launch system. Originally, they were to launch on the smaller Falcon 1e rocket.
      </td>
      <td>
        <span
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#36454F",
              "border": "1px solid gray",
              "color": "white",
              "display": "block",
              "width": "120px",
            }
          }
        >
           
          Not a Member
        </span>
      </td>
      <td>
        <button
          className="joinBtn"
          onClick={[Function]}
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
    <tr
      style={
        Object {
          "backgroundColor": "#E5E4E2",
        }
      }
    >
      <td
        className="name"
      >
        ABS
      </td>
      <td
        className="description"
      >
        ABS, formerly Asia Broadcast Satellite, is a global satellite operator based in Hong Kong and officially incorporated in Bermuda. Its services include direct-to-home and satellite-to-cable TV distribution, cellular services, and internet services. Operating 6 communication satellites, the satellite fleet currently covers 93% of the world’s population including the Americas, Africa, Asia Pacific, Europe, the Middle East, Russia and Commonwealth of Independent States.
      </td>
      <td>
        <span
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#36454F",
              "border": "1px solid gray",
              "color": "white",
              "display": "block",
              "width": "120px",
            }
          }
        >
           
          Not a Member
        </span>
      </td>
      <td>
        <button
          className="joinBtn"
          onClick={[Function]}
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
    <tr
      style={
        Object {
          "backgroundColor": "#E5E4E2",
        }
      }
    >
      <td
        className="name"
      >
        Eutelsat
      </td>
      <td
        className="description"
      >
        Eutelsat S.A. is a European satellite operator. Providing coverage over the entire European continent, the Middle East, Africa, Asia and the Americas, it is the world's third largest satellite operator in terms of revenues.
      </td>
      <td>
        <span
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#36454F",
              "border": "1px solid gray",
              "color": "white",
              "display": "block",
              "width": "120px",
            }
          }
        >
           
          Not a Member
        </span>
      </td>
      <td>
        <button
          className="joinBtn"
          onClick={[Function]}
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
